docs/opcodes_details.md

```diff
@@ -342,4 +356,137 @@ can be dropped
+|----------|----------------|-----------------------------------|-------------------------------------|-------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------|
+| 0x00 | STOP           | Zero    |               | O(1) | O(1) | O(1) |
+| 0x01 | ADD            | Verylow | x + y         | O(1) | O(1) | O(1) |
+| 0x02 | MUL            | Low     | x * y         | O(1) | O(num_words(x)) | O(1) |
+| 0x03 | SUB            | Verylow | x - y         | O(1) | O(1) | O(1) |
+| 0x04 | DIV            | Low     | x / y         | depends on x:<br />- x is 64bit: O(1)<br />- else: full Knuth | depends on y:<br />- y is 64 bit: O(num_words(x))<br />- y is 128 bit: (2 * num_words(x))<br />- else: full Knuth   | bitwise long division: O(length(y)^2)? |
+| 0x05 | SDIV           | Low     | x / y         | same as above | same as above | same as above |
+| 0x06 | MOD            | Low     | x % y         | same as above | same as above | same as above |
+| 0x07 | SMOD           | Low     | x % y         | same as above | same as above | same as above |
+| 0x08 | ADDMOD         | Mid     |               | same as above | same as above | same as above |
+| 0x09 | MULMOD         | Mid     |               | same as above | same as above | same as above |
+| 0x0a | EXP            |         | x ^ n         | O(num_words(n) * n * log x)? | O(n * log x)? | O(n * log x)? |
+| 0x0b | SIGNEXTEND     | Low     |               | O(1) | O(1) | O(1) |
+| 0x10 | LT             | Verylow |               | O(1) | O(1) | O(1) |
+| 0x11 | GT             | Verylow |               | O(1) | O(1) | O(1) |
+| 0x12 | SLT            | Verylow |               | O(1) | O(1) | O(1) |
+| 0x13 | SGT            | Verylow |               | O(1) | O(1) | O(1) |
+| 0x14 | EQ             | Verylow |               | O(1) | O(1) | O(1) |
+| 0x15 | ISZERO         | Verylow |               | O(1) | O(1) | O(1) |
+| 0x16 | AND            | Verylow |               | O(1) | O(1) | O(1) |
+| 0x17 | OR             | Verylow |               | O(1) | O(1) | O(1) |
+| 0x18 | XOR            | Verylow |               | O(1) | O(1) | O(1) |
+| 0x19 | NOT            | Verylow |               | O(1) | O(1) | O(1) |
+| 0x1a | BYTE           | Verylow |               | O(1) | O(1) | O(1) |
+| 0x1b | SHL            | Verylow | x << n        | depends on n:<br />- n > 192: O(1)<br />- n > 128: O(3)<br />- n > 64: O(5)<br />- else: O(7)   | depends on n:<br />- n < 64: O(1)<br />- n < 128: O(2)<br />- else: O(3) | O(num_words(x)) |
+| 0x1c | SHR            | Verylow | x >> n        | same as above | same as above | same as above |
+| 0x1d | SAR            | Verylow | x >> n        | same as above | same as above | same as above |
+| 0x30 | ADDRESS        |         |               | | | |
```

Comment:
lets drop the empty remainder of the table, I don't think complexity is relevant to any of these, and if it is e.g. for memory ops, it's already described above